ngraph/core/src/op/non_zero.cpp

```diff
@@ -75,7 +75,9 @@ void op::v3::NonZero::validate_and_infer_types()
     }
     else
     {
-        set_output_type(0, m_output_type, PartialShape{input_shape.rank(), Dimension::dynamic()});
+        const Dimension dim =
+            input_shape.is_static() ? shape_size(input_shape.get_shape()) : Dimension::dynamic();
```

Comment:
Could you please manually multiply Dimensions and skip this restriction of shape_size?

Reply:
Done.